astroid/decorators.py

```diff
@@ -89,15 +89,17 @@ def __get__(self, inst, objtype=None):
         return val
 
 
-def path_wrapper(func):
+def path_wrapper(func: Callable) -> Callable:
```

Comment:
This should probably be a `Typevar` or even a `ParamSpec`.

Reply:
Could you explain more? I haven't used those.

Reply:
I think the PEP introducing `ParamSpec` is quite clear:
https://peps.python.org/pep-0612/

It explains its uses and those can be applied here.

Reply:
That sounds out of scope for this PR. I can undo this change if you want.

Reply:
Yeah, let's revert then